aten/src/ATen/core/SymbolicIntNode.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
+#include <vector>
+#include <ATen/core/SymInt.h>
+#include <mutex>
+
+namespace c10 {
+
+class TORCH_API SymbolicIntNode {
+    public:
+        c10::SymInt toSymInt();
+        static bool is_symbolic(int64_t data) {
+            return SYM_TAG_MASK & data;
+        }
+
+        const static int64_t SYM_TAG_MASK = 1LL << 63;
+
+        virtual ~SymbolicIntNode() {};
+    private:
+        int64_t data_;
+        
+};
+
+class TORCH_API SymIntTable {
+
+public:
+    
+    size_t addNode(SymbolicIntNode* sin);
+    SymbolicIntNode* getNode(size_t index);
+
+    ~SymIntTable() {
+        for (auto sit: nodes_) {
+            free(sit);
```

Comment:
instead of doing this make the vector store an owning pointer (either `unique_ptr` or `shared_ptr`) so that you don't need an explicit destructor

Reply:
Yeah, I was thinking we could save some overhead here, but it may not worth it.

Reply:
`unique_ptr` is supposed to be zero overhead on top of a regular raw pointer (due to ABI shenanigans this is not strictly true but you are mostly definitely not thinking about that here.)